lib/runtime-core/src/fault.rs

```diff
@@ -615,30 +615,46 @@ pub unsafe fn get_fault_info(siginfo: *const c_void, ucontext: *mut c_void) -> F
     known_registers[X64Register::GPR(GPR::RBP).to_index().0] = Some(gregs[REG_RBP as usize] as _);
     known_registers[X64Register::GPR(GPR::RSP).to_index().0] = Some(gregs[REG_RSP as usize] as _);
 
-    if !(*ucontext).uc_mcontext.fpregs.is_null() {
-        let fpregs = &*(*ucontext).uc_mcontext.fpregs;
-        known_registers[X64Register::XMM(XMM::XMM0).to_index().0] = Some(read_xmm(&fpregs._xmm[0]));
-        known_registers[X64Register::XMM(XMM::XMM1).to_index().0] = Some(read_xmm(&fpregs._xmm[1]));
-        known_registers[X64Register::XMM(XMM::XMM2).to_index().0] = Some(read_xmm(&fpregs._xmm[2]));
-        known_registers[X64Register::XMM(XMM::XMM3).to_index().0] = Some(read_xmm(&fpregs._xmm[3]));
-        known_registers[X64Register::XMM(XMM::XMM4).to_index().0] = Some(read_xmm(&fpregs._xmm[4]));
-        known_registers[X64Register::XMM(XMM::XMM5).to_index().0] = Some(read_xmm(&fpregs._xmm[5]));
-        known_registers[X64Register::XMM(XMM::XMM6).to_index().0] = Some(read_xmm(&fpregs._xmm[6]));
-        known_registers[X64Register::XMM(XMM::XMM7).to_index().0] = Some(read_xmm(&fpregs._xmm[7]));
-        known_registers[X64Register::XMM(XMM::XMM8).to_index().0] = Some(read_xmm(&fpregs._xmm[8]));
-        known_registers[X64Register::XMM(XMM::XMM9).to_index().0] = Some(read_xmm(&fpregs._xmm[9]));
-        known_registers[X64Register::XMM(XMM::XMM10).to_index().0] =
-            Some(read_xmm(&fpregs._xmm[10]));
-        known_registers[X64Register::XMM(XMM::XMM11).to_index().0] =
-            Some(read_xmm(&fpregs._xmm[11]));
-        known_registers[X64Register::XMM(XMM::XMM12).to_index().0] =
-            Some(read_xmm(&fpregs._xmm[12]));
-        known_registers[X64Register::XMM(XMM::XMM13).to_index().0] =
-            Some(read_xmm(&fpregs._xmm[13]));
-        known_registers[X64Register::XMM(XMM::XMM14).to_index().0] =
-            Some(read_xmm(&fpregs._xmm[14]));
-        known_registers[X64Register::XMM(XMM::XMM15).to_index().0] =
-            Some(read_xmm(&fpregs._xmm[15]));
+    // Skip reading floating point registers when building with musl libc.
+    // FIXME: Depends on https://github.com/rust-lang/libc/pull/1646
+    #[cfg(not(target_env = "musl"))]
```

Comment:
Do we need to handle the other case too?  Like

```
#[cfg(target_env = "musl")]
unimplemented!("blocked on https://github.com/rust-lang/libc/pull/1646");
```

Or will it work fine as-is?

Reply:
This is on the always-taken path for fault handling so panicking here will break things that depend on fault handling, like middlewares. Now all XMM registers will be defaulted to zero if not dumped; do you think it is fine?

Reply:
Hmm, I'm not familiar enough to know what the impacts of that are: if you think this leaves the system in a sensible state where execution makes sense then I'm happy with it!  Otherwise, I don't think compiling and failing silently is necessarily a step up from panicking or not compiling.

Reply:
The only case where doing so is possibly incorrect is when we take a managed snapshot and then unwind out from WebAssembly code (where floating point state will be lost). This feature is not widely used (yet) and I think it is "cold" enough to ignore for now and defer to after a future upstream fix in musl-libc.

Reply:
Okay, that sounds reasonable to me!  We just need to remember to not forget to update this!